exercises/affine-cipher/description.md

```diff
@@ -0,0 +1,54 @@
+Create an implementation of the affine cipher,
+an ancient encryption system created in the Middle East.
+ 
+The affine cipher is a type of monoalphabetic substitution cipher.
+Each character is mapped to its numeric equivalent, encrypted with
+a mathematical function and then converted to the letter relating to
+its new numeric value. Although all monoalphabetic ciphers are weak,
+the affine cypher is much stronger than the atbash cipher,
+because it has many more keys.
+ 
+the encryption function is:
+ 
+  **E(x) = (ax + b) % m**
+  -  where `x` is the letter's index from 0 - length of alphabet - 1
+  -  `m` is the length of the alphabet - roman alphabet = 26
+  -  and `a` and `b` make the key
+ 
+the decryption function is:
+ 
+  **D(y) = a^-1(y - b) % m**
+  -  where `y` is the numeric value of an encrypted letter, ie. y = E(x)
+  -  it is important to note that a^-1 is the modal multiplicative inverse
+  -  the modal multiplicative inverse of a only exists if `a` and `m` are
+  -  coprime.
+ 
+To find the MMI of `a`:
+
+  **1 = an % m**
+  -  where `n` is the modal multiplicative inverse of `a` mod `m`
```

Comment:
This appears to be a recursive definition. In pseudocode:

```python
def mmi(a, m):
    return a * mmi(a % m) % m
```

Is this meant to be a recursive computation? If not, then I'd like to see some expansion of how to compute the MMI of a value.

Reply:
Checking the MMI in ruby is as simple as
```ruby
  raise ArgumentError, "Error: a and m must be coprime!" if a.gcd(26) != 1
I could change the `%` in the description to `mod`.  It is literally just the math for finding MMI.

Reply:
- I could change the `%` in the description to `mod`.  It is literally just the math for finding MMI.

Reply:
I agree it is nice to be clear in the README.md's but I also want learners to search the internet to learn more about the math that is going on.  When I built this exercise I didn't know what a `modal multiplicative inverse` was.  I think that learning through self driven research helps us learn how to solve problems better.

Reply:
In ruby the MMI can be found with something like this:
```ruby
  inverse = 0
  (1..25).each {|i| inverse = i if (@a * i) % 26 == 1}

Reply:
#### To be clear:
- I think that it is beneficial for most learners to implement a solution which includes a full decryption and finds the `MMI`
- For a fast, resource light implementation is better to revise the iteration so that:
  - the alphabet is encrypted with the function `E(x) = (ax + b) mod m`
  - all encoding and decoding simply translates between the order of the alphabet and its encrypted counterpart.

Reply:
I strongly disagree about the desirability of forcing learners to search the internet for information required to solve the problem. When I was going through the exercism exercises for the first time, I was frustrated to the point of turning in bogus unimplemented solutions with certain of the exercises which required knowledge of that sort. I was a motivated learner, but I wanted to learn about programming in a new language, not about the rules by which nucleotides bind to each other.

Using `%` for `mod` is fine; that's not the issue. The issue is that I don't know what the modal multiplicative inverse is, and the definition provided here is brief to the point of incomprehensibility.  This exercise should include a precise and useful definition of what exactly the MMI is, because students aren't coming to the exercise to learn about the MMI, they're coming here to learn the language of their choice.